### PR TITLE
🌱 observability: move metrics to config and use sidecar in kube-state-metrics

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

This PR:
* moves the metrics configuration from `hack/observability/kube-state-metrics` to `config/metrics`.
* adds a ClusterRole in `config/metrics/crd-clusterrole.yaml` for the aggregated clusterrole for kube-state-metrics
* adds a `kustomization.yaml` to `config/metrics` which includes the above two
* Changes to the kustomization of kube-state-metrics:
  * Adds a patch to add a sidecar container to kube-state-metrics which
    * periodicly lists configmaps in namespace `observability` which have the label `kube-state-metrics/custom-resource`  set
    * combines the yamls of the configmaps to a single configuration file for kube-state-metrics
    * updates the config file for kube-state-metrics so it reloads the configuration automatically
  * Adds a aggregated clusterrole + clusterrole binding so providers can dynamically add additional rbac rules to allow kube-state-metrics to watch the relevant resources
* Changes to `tilt up`:
  * Adds a configuration option `deploy_kustomizations` which allows to define paths to additional kustomizations which then get deployed to the cluster
    * Example usage: to reference  a kustomization which adds a configuration + clusterrole for additional kube-state-metrics crd configuration.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

/area observability

### CODE REVIEW

File: hack/observability/kube-state-metrics/crd-sidecar-patch.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      - command:
+        - /bin/sh
+        - -c
+        - |
+          cat << EOF > "/etc/config/crd-config.yaml"
+          kind: CustomResourceStateMetrics
+          spec:
+            resources: []
+          EOF
+        image: kiwigrid/k8s-sidecar:latest
```
Comment (sbueringer) [inline hack/observability/kube-state-metrics/crd-sidecar-patch.yaml:37]: What is the ~ featureset of this image? Let's add a comment

File: hack/observability/kube-state-metrics/kustomization.yaml [hunk 2]
```diff
@@ -1,15 +1,30 @@
 resources:
   - ../namespace.yaml
-  # The kube-state-metrics helm chart will reference a configmap with name `kube-state-metrics-crd-config`.
-  # The configMapGenerator below will create the configmap and append a hash suffix calculated from its
-  # content to the name. Kustomize will append the suffix hash to all references in the helm chart, but
-  # only when the helm chart content is referenced in "resources".
-  # This would not work if the helm chart is configured in this file via the "helmCharts" option.
-  - ./chart
+  # Aggregated ClusterRole and ClusterRoleBinding for kube-state-metrics crd resources.
+  - rbac-crd-aggregation.yaml
```
Comment (sbueringer) [inline hack/observability/kube-state-metrics/kustomization.yaml:4]: Why do we add this and the next file here and in chart/kustomization.yaml?

File: hack/observability/kube-state-metrics/kustomization.yaml [hunk 3]
```diff
@@ -1,15 +1,30 @@
...
+  # Adds the core CAPI crd metrics configmap and CusterRole.
+  - ../../../config/metrics
 
 namespace: observability
 
-configMapGenerator:
-- name: kube-state-metrics-crd-config
-  files:
-  - crd-config.yaml
+helmCharts:
```
Comment (sbueringer) [inline hack/observability/kube-state-metrics/kustomization.yaml:12]: Why do we have helmCharts twice as well?

**Review by sbueringer — COMMENTED** [review]
Nice one, just a few smaller findings

File: docs/book/src/developer/tilt.md [hunk 4]
```diff
@@ -214,6 +214,14 @@ Supported values are:
 
 \*: Note: the UI will be accessible via a link in the tilt console
 
+**deploy_kustomizations** (map[string]string, default={}): If set, installs the additional kustomizations to the cluster.
+
+Example:
+```yaml
+deploy_kustomizations:
+  capv-metrics: /path/to/kustomize/directory/config/metrics
```
[the remark this answers is not in the document]
Comment (chrischdi) [reply docs/book/src/developer/tilt.md:222]: yes it works with relative paths 👍

File: hack/observability/kube-state-metrics/kustomization.yaml (hunk 3 above)
Comment (chrischdi) [reply hack/observability/kube-state-metrics/kustomization.yaml:12]: I just forgot to remove `chart/kustomization.yaml`

**Comment by sbueringer** [comment]
Thank you!

/lgtm
/approve

**Pull request merged**